src/Serokell/Data/Variant/Serialization.hs

```diff
@@ -87,6 +92,39 @@ instance Aeson.FromJSON Variant where
 -- `decode . encode` should be `id`.
 
 -- TODO: move it somewhere??
+
+genericGetVectorWith :: (VG.Vector v a) => Cereal.Get a -> Cereal.Get (v a)
+{-# INLINE genericGetVectorWith #-}
+genericGetVectorWith getter = do
+    len64 <- Cereal.get :: Cereal.Get Int64
+    when (len64 > fromIntegral (maxBound :: Int)) $
+        fail "Host can't deserialize a Vector longer than (maxBound :: Int)"
+    VG.replicateM (fromIntegral len64) getter
+
+-- | Write a 'Data.Vector.Generic.Vector' using custom 'Putter' for
+-- the vector's elements.
+genericPutVectorWith :: (VG.Vector v a) => Cereal.Putter a -> Cereal.Putter (v a)
+{-# INLINE genericPutVectorWith #-}
+genericPutVectorWith putter v = do
+    Cereal.put ((fromIntegral $ VG.length v) :: Int64)
+    VG.mapM_ putter v
+
+-- | Write a 'Data.Vector.Generic.Vector'.
+genericPutVector :: (Cereal.Serialize a, VG.Vector v a) => Cereal.Putter (v a)
+{-# INLINE genericPutVector #-}
+genericPutVector = genericPutVectorWith Cereal.put
+
+-- | Read a 'Data.Vector.Generic.Vector'.
+genericGetVector :: (Cereal.Serialize a, VG.Vector v a) => Cereal.Get (v a)
+{-# INLINE genericGetVector #-}
+genericGetVector = genericGetVectorWith Cereal.get
+
+instance (Cereal.Serialize a) => Cereal.Serialize (V.Vector a) where
```

Comment:
I think that it's better to import orphan instances from their respective packages than to put them into serokell-util.